server/shared/src/main/scala/org/http4s/server/middleware/GZip.scala

```diff
@@ -89,4 +106,16 @@ object GZip extends GZipPlatform {
       .putHeaders(`Content-Encoding`(ContentCoding.gzip))
       .pipeBodyThrough(compressPipe)
   }
+
+  private def unzipRequest[F[_]: Compression](
+      bufferSize: Int,
+      req: Request[F],
+  ): Request[F] = {
+    val decompressPipe = Compression[F].gunzip(bufferSize = bufferSize).andThenF(_.content)
+    logger.trace("GZip middleware decoding content").unsafeRunSync()
```

Comment:
I don't love running this effect like this, though I see that it is status quo. 

Perhaps in 1.x we should move the logging out to a separate wrapper method. Something like,

```scala

object Gzip extends GzipPlatform {

  def withLogging[F[_]: Functor, G[_]: Compression: Sync](...):` Http[F, G]
  ```

Reply:
See:
- https://github.com/http4s/http4s/pull/6645

Reply:
Ah I see. Makes sense.

Reply:
Changing existing code is out-of-scope. Avoiding writing new code with anti-pattterns (established as the may be) is in-scope :)

Reply:
Got it! I'll change it a little

Reply:
> I couldn't provide enough evidence to create `GZip` middleware.

What's the error?

Reply:
If I change the signature of `Gzip::apply` to something like `def apply[F[_]: LoggerFactory: Monad, G[_]: Compression](...)` then the arguments of `GZipPlatform::apply` must change as well. Currently it's

```
def apply[F[_], G[_]](
      http: Http[F, G],
      bufferSize: Int,
      level: DeflateParams.Level,
      isZippable: Response[G] => Boolean,
      F: Functor[F],
      G: Sync[G],
  ): Http[F, G]
```
But `F: Functor[F]` is not enough. I tried various approaches but can't find anything that would work.